src/sentry/mail/adapter.py

```diff
@@ -85,9 +84,8 @@ def get_sendable_user_objects(project):
         Return a collection of USERS that are eligible to receive
         notifications for the provided project.
         """
-        return NotificationSetting.objects.get_notification_recipients(project)[
-            ExternalProviders.EMAIL
-        ]
+        recipients = NotificationSetting.objects.get_notification_recipients(project)
+        return set.union(*recipients.values()) if recipients else {}
```

Comment:
`get_notification_recipients()` never returns `None` so it's safe to always return the union. I'd also rename the variable to make it clearer why we're calling `.values()`.
```python
recipients_by_provider = NotificationSetting.objects.get_notification_recipients(project)
return set.union(*recipients_by_provider.values())
```

Reply:
It does in https://github.com/getsentry/sentry/blob/36cb56e1dc995cdb79d5f76f137527a1d364046f/tests/sentry/mail/test_adapter.py#L759

which is why I added the check

Reply:
I just checked this out:
```
>>> set.union(set([1]), set([1]))
{1}
>>> set.union(set([1]), [1])
{1}
>>> set.union([1], [1])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: descriptor 'union' requires a 'set' object but received a 'list'
```
I think because `.values()` returns lists you won't be able to use `set.union(*...)`.
```python
recipients_by_provider = NotificationSetting.objects.get_notification_recipients(project)
return set([user for users in recipients_by_provider.values() for user in users])
```